### PR TITLE
fix: Solana Native and Steel Action should only work on changes.

### PR DESCRIPTION
**Pull request by heyAyushh**

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -43,7 +43,7 @@ Specifically for code in this repo:
    [Biome](https://biomejs.dev/). Execute the following command to format and lint your code at the root of this project before submitting a pull request:
 
 ```bash
-pnpm check:fix
+pnpm fix
```
Comment (jacobcreech) [inline CONTRIBUTING.md:46]: Did you update package.json?

File: .github/workflows/solana-native.yml [hunk 2]
```diff
@@ -11,107 +11,246 @@ on:
...
+        uses: heyAyushh/setup-solana@v5.5
+        with:
+          solana-cli-version: stable
+      - name: Build and Test with Stable
+        run: |
+          source build_and_test.sh
+          solana -V
+          rustc -V
+          process_projects "stable"
+      - name: Setup Solana 1.18.17
```
Comment (Woody4618) [inline .github/workflows/solana-native.yml:192]: Why not use the solana version of the project?

File: CONTRIBUTING.md (hunk 1 above)
Comment (heyAyushh) [reply CONTRIBUTING.md:46]: The command was wrong, package.json only contains [fix script](https://github.com/solana-developers/program-examples/blob/45e302abb5213363fef5f4db9defbcf51a5c794a/package.json#L10)

**Review by Woody4618 — APPROVED** [review]

File: .github/workflows/solana-native.yml [hunk 3]
```diff
@@ -11,107 +11,233 @@ on:
...
+              - added|modified: '**/native/**'
+            workflow:
+              - added|modified: '.github/workflows/solana-native.yml'
+      - name: Analyze Changes
+        id: analyze
         run: |
-          npm install --global pnpm
-      - name: Build Native programs
+          # Generate ignore pattern, excluding comments
+          ignore_pattern=$(grep -v '^#' .github/.ghaignore | grep -v '^$' | tr '\n' '|' | sed 's/|$//')
```
Comment (Woody4618) [inline .github/workflows/solana-native.yml:44]: Is this ghaignore file always there in these projects?

File: .github/workflows/solana-native.yml (hunk 2 above)
Comment (Woody4618) [reply .github/workflows/solana-native.yml:192]: You could use verified builds for this maybe. It has docker images for all solana versions

File: .github/workflows/solana-native.yml (hunk 3 above)
Comment (heyAyushh) [reply .github/workflows/solana-native.yml:44]: it's in the github actions folder, to ignore specific projects. always there 🫡

File: .github/workflows/solana-native.yml (hunk 2 above)
Comment (heyAyushh) [reply .github/workflows/solana-native.yml:192]: This 1.18.17 version is the minimum supported solana version for this repository, stable is to track upcoming bugs or failures.

Comment (heyAyushh) [reply .github/workflows/solana-native.yml:192]: Verified builds should be another PR I think.

Comment (Woody4618) [reply .github/workflows/solana-native.yml:192]: Yeah for sure. Just an idea

**Pull request merged**